ui/src/main/webapp/config/webpack.prod.js

```diff
@@ -3,36 +3,33 @@ var webpackMerge = require('webpack-merge');
 var ExtractTextPlugin = require('extract-text-webpack-plugin');
 var commonConfig = require('./webpack.common.js');
 var helpers = require('./helpers');
+var ngtools = require('@ngtools/webpack');
+var AotPlugin = ngtools.AotPlugin;
 
 const ENV = process.env.NODE_ENV = process.env.ENV = 'production';
 
 module.exports = webpackMerge(commonConfig, {
     devtool: 'source-map',
 
     output: {
-        path: helpers.root('dist'),
-        publicPath: '/',
-        filename: 'js/[name].[hash].js',
-        chunkFilename: 'js/[id].[hash].chunk.js'
+        path: helpers.root('../../../target/windup-web'),
+        filename: 'js/[name].js',
+        chunkFilename: 'js/[id].chunk.js'
     },
-/*
-    htmlLoader: {
-        minimize: false // workaround for ng2
-    },
-*/
+
+
     plugins: [
-        new webpack.NoErrorsPlugin(),
-//        new webpack.optimize.DedupePlugin(),
+        new webpack.NoEmitOnErrorsPlugin(),
+        new ExtractTextPlugin('css/[name].css'),
+        new AotPlugin({
+            tsConfigPath: './tsconfig-production.json',
+            basePath: '.',
+            mainPath: 'src/main.ts'
+        }),
         new webpack.optimize.UglifyJsPlugin({ // https://github.com/angular/angular/issues/10618
             mangle: {
                 keep_fnames: true
             }
-        }),
-        new ExtractTextPlugin('[name].[hash].css'),
-        new webpack.DefinePlugin({
-            'process.env': {
-                'ENV': JSON.stringify(ENV)
```

Comment:
I think this is ugly workaround which was required because webpack wouldn't set env variable properly without that.
It may have been fixed since I tried it for the last time.